ambari-agent/conf/unix/agent-multiplier.py

```diff
@@ -226,7 +226,7 @@ def create_host_name_script(self, host_name, host_name_script):
                "echo HOSTNAME"
     with open(str(host_name_script), "w+") as f:
       f.writelines(template.replace("HOSTNAME", host_name))
-    subprocess32.call("chmod +x %s" % host_name_script, shell=True)
+    subprocess.call("chmod +x %s" % host_name_script, shell=True)
```

Comment:
@lingarajg  

I believe that the [subprocess32](https://github.com/apache/ambari/blob/trunk/ambari-common/src/main/python/ambari_commons/subprocess32.py) module is a copy (maybe there is some customization but I don't know) of Python's standard subprocess module, so the most viable thing would be to maintain the use of the module that is already standard in the language and not maintain a "separate" module, in my opinion it would not make much sense.

As for the use of shell=True, it should be avoided taking into account the issue of security and the possibility of executing some arbitrary code.
_shell=True_ It is sometimes convenient to make use of shell-specific features such as word splitting or parameter expansion.

We would have a syntax change at all points where shell=True is executed, example:

of this

`subprocess.Popen("command -with -options 'like this' and\\ an\\ argument", shell=True)
`

For that
`subprocess.Popen(['command', '-with','-options', 'like this', 'and an argument'])`

Reply:
I agree with you that Python introduces true concurrency through the use of multiple processes. However, Python 2 doesn't have the subprocess module by default. Therefore, we copied the subprocess module from Python 3's standard library and made compatibility modifications to maintain it within the Ambari codebase. So, if we upgrade to Python 3, we won't need to maintain our own version of the subprocess module anymore and can directly use the one provided by Python 3.